### PR TITLE
Feature/genome factory hive module

### PR DESCRIPTION
**Pull request by vinay-ebi**
Hive Modules for genomeFactory & datasetFactory

### CODE REVIEW

File: src/python/ensembl/production/factories/metadata/genomeFactory.py [hunk 1]
```diff
@@ -20,451 +20,524 @@
...
+from sqlalchemy import JSON
 from ensembl.production.metadata.api.models.genome import Genome, GenomeDataset
 from ensembl.production.metadata.api.models.organism import Organism, OrganismGroup, OrganismGroupMember
 from ensembl.production.metadata.api.models.assembly import Assembly
-from ensembl.production.metadata.api.models.dataset import DatasetType, Dataset, DatasetSource
-
-logging.basicConfig(level=logging.INFO, format='%(message)s')
-logger = logging.getLogger(__name__)
+from ensembl.production.metadata.api.models.dataset import DatasetType, Dataset, DatasetSource, DatasetStatus
+from ensembl.production.metadata.api.hive.dataset_factory import DatasetFactory
```
Comment (marcoooo) [inline src/python/ensembl/production/factories/metadata/genomeFactory.py:30]: I suspect this dependency path is not available any more. following latest updates in api code structure.

**Review by marcoooo — COMMENTED** [review]
I bet there are things to do following other PR here: https://github.com/Ensembl/ensembl-metadata-api/pull/75/ Am I right?

File: src/python/ensembl/production/factories/metadata/genomeFactory.py (hunk 1 above)
Comment (vinay-ebi) [reply src/python/ensembl/production/factories/metadata/genomeFactory.py:30]: yes, changes are needed for the genome factory and hive module, I will  update accordingly

Comment (marcoooo) [reply src/python/ensembl/production/factories/metadata/genomeFactory.py:30]: Wait for the next tag which should arrive today the. Thanks!

File: src/python/ensembl/production/hive/HiveDatasetFactory.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+            )
+
+        except KeyError as error:
+            raise KeyError(f"Missing request parameters: {str(error)}")
+
+        except Exception as e:
+            raise Exception(f"An unexpected error occurred: {str(e)}")
+
+
+def write_output(self):
```
Comment (marcoooo) [inline src/python/ensembl/production/hive/HiveDatasetFactory.py:93]: Indentation issue here.

**Review by marcoooo — CHANGES_REQUESTED** [review]

File: modules/Bio/EnsEMBL/Production/Pipeline/PipeConfig/Base_conf.pm [hunk 3]
```diff
@@ -88,9 +158,10 @@ sub resource_classes {
 
     my %output = (
         #Default is a duplicate of 100M
-        'default'   => { 'LSF' => '-q ' . $self->o('production_queue'), 'SLURM' => $pq . $time{'H'} . ' --mem=' . $memory{'100M'} . 'm' },
-        'default_D' => { 'LSF' => '-q ' . $self->o('production_queue'), 'SLURM' => $pq . $time{'D'} . ' --mem=' . $memory{'100M'} . 'm' },
-        'default_W' => { 'LSF' => '-q ' . $self->o('production_queue'), 'SLURM' => $pq . $time{'W'} . ' --mem=' . $memory{'100M'} . 'm' },
+        'ensemblgenome_4GB' => {'LSF' => '-q '.$self->o('production_queue').' -M 4000 -R "rusage[mem=4000]"'},
```
Comment (marcoooo) [inline modules/Bio/EnsEMBL/Production/Pipeline/PipeConfig/Base_conf.pm:161]: Why this difference in naming?

File: src/python/ensembl/production/factories/metadata/genomeFactory.py [hunk 4]
```diff
@@ -20,451 +20,536 @@
...
 from sqlalchemy import select
-from sqlalchemy import update
-
-from ensembl.database import DBConnection
+from sqlalchemy import JSON
 from ensembl.production.metadata.api.models.genome import Genome, GenomeDataset
 from ensembl.production.metadata.api.models.organism import Organism, OrganismGroup, OrganismGroupMember
 from ensembl.production.metadata.api.models.assembly import Assembly
 from ensembl.production.metadata.api.models.dataset import DatasetType, Dataset, DatasetSource
+from ensembl.production.metadata.api.factories.datasets import DatasetFactory
```
Comment (marcoooo) [inline src/python/ensembl/production/factories/metadata/genomeFactory.py:30]: No import of GenomeFactory from https://github.com/Ensembl/ensembl-metadata-api/blob/main/src/ensembl/production/metadata/api/factories/genome.py ?

**Pull request merged**